pkg/serverless/appsec/httpsec/proxy.go

```diff
@@ -206,14 +208,24 @@ func (lp *ProxyLifecycleProcessor) spanModifier(lastReqId string, chunk *pb.Trac
 		)
 
 	case *events.ALBTargetGroupRequest:
+		var sourceIp string
+		// ALB is a "trusted" origin, so we extract the source IP from the X-Forwarded-For header.
+		// ALB also provides headers with fully lower-case names, so we don't have to worry about case.
+		if xff, found := event.MultiValueHeaders["x-forwarded-for"]; found && len(xff) > 0 {
+			sourceIp = xff[0]
+		} else if xff, found := event.Headers["x-forwarded-for"]; found {
+			sourceIp = xff
+		}
```

Comment:
I think you can remove this part: "Source IP" is supposed to be the connection IP address (to the TCP socket) in the the `setClientIPTags` function, and the XFF header is already taken into account by the following call to `setClientIPTags(span, ctx.requestSourceIP, reqHeaders)`